File: WEEK1/DAY4/Prog5/P5.py
def numDigit(N):
    l=0
    while N>0:
        l+=1
        N//=10
    return l
def armstrong(N):
    l=numDigit(N)
    ans=0
    n=N
    while(n>0):
        dig=n%10
        ans+=(dig**l)
        n//=10
    if ans==N:
        print("Armstrong number")
    else:
        print("Not armstrong number")

if __name__=='__main__':
    print("Check if N is an Armstrong number")
    N=int(input("Enter N: "))
    armstrong(N)

File: WEEK1/DAY6/Prog5/P5.c
#include <stdio.h>
int factorial(int n){
    if(n==1 || n==0){
        return 1;
    }
    return n*factorial(n-1);
}
int main(){
    int N;
    printf("Check if N is strong number\n");
    printf("Enter N: ");
    scanf("%d",&N);
    int n=N;
    int dig;
    int ans=0;
    while(n>0){
        dig=n%10;
        ans+=factorial(dig);
        n/=10;
    }
    if(N==ans){
        printf("Strong number\n");
    }
    else{
        printf("Not strong number\n");
    }
}